### ISSUE TITLE
NPE in DDLambda v0.2.1

### ISSUE THREAD

**Issue by francis-a** [DataDog/datadog-lambda-java#34]
In version 0.2.1 it looks like there is a NPE coming from `DDLambda.init`. I'm not 100% sure but it looks like it could be cause by this change:
https://github.com/DataDog/datadog-lambda-java/compare/0.2.0...0.2.1

My invocation log line doesn't seem to contain a `function_arn` and this looks like it would cause the NPE. I'm not sure if the root cause here is that `function_arn` is required or if this code needs to be implemented with null safety. 

```
{
    "m": "aws.lambda.enhanced.invocations",
    "v": 1,
    "t": [
        "memorysize:4096",
        "account_id:xxxxx",
        "resource:xxxxx",
        "functionname:xxxxx",
        "runtime:java11.0.9.1",
        "region:eu-west-1",
        "datadog_lambda:0.2.1",
        "cold_start:false"
    ],
    "e": 1614949078
}
```

Stack trace:
```
java.lang.NullPointerException: java.lang.NullPointerExceptionjava.lang.NullPointerException
at io.opentracing.propagation.TextMapExtractAdapter.iterator(TextMapExtractAdapter.java:38)
at datadog.trace.bootstrap.instrumentation.api.ContextVisitors$EntrySetContextVisitor.forEachKey(ContextVisitors.java:55)
at datadog.trace.bootstrap.instrumentation.api.ContextVisitors$EntrySetContextVisitor.forEachKey(ContextVisitors.java:49)
at datadog.trace.agent.core.propagation.TagContextExtractor.extract(TagContextExtractor.java:27)
at datadog.trace.agent.core.propagation.HttpCodec$CompoundExtractor.extract(HttpCodec.java:97)
at datadog.trace.agent.core.CoreTracer.extract(CoreTracer.java:405)
at datadog.trace.instrumentation.opentracing32.OTTracer.extract(OTTracer.java:68)
at io.opentracing.util.GlobalTracer.extract(GlobalTracer.java:204)
at com.datadoghq.datadog_lambda_java.DDLambda.startSpan(DDLambda.java:139)
at com.datadoghq.datadog_lambda_java.DDLambda.<init>(DDLambda.java:53)
```

**Comment by seanstaley** [comment]
I'm encountering the same issue with 0.2.0. In my case I am passing simply the Lambda context into the constructor:

```java
var datadog = new DDLambda(awsContext);
```

**Comment by seanstaley** [comment]
I was able to fix this issue in my environment by passing in an empty map instead of `null` into the `startSpan()` method.

```java
    /**
     * Create a new DDLambda instrumenter given some Lambda context
     *
     * @param cxt Enhanced Metrics pulls information from the Lambda context.
     */
    public DDLambda(Context cxt) {
        this.tracing = new Tracing();
        this.enhanced = checkEnhanced();
        recordEnhanced(INVOCATION, cxt);
        addTraceContextToMDC();
        startSpan(new HashMap<>(), cxt);
    }

    /**
     * Testing only: create a DDLambda instrumenter with a given context and xrayTraceInfo
     *
     * @param cxt           Enhanced Metrics pulls information from the Lambda context.
     * @param xrayTraceInfo This would normally be the contents of the "_X_AMZN_TRACE_ID" env var
     */
    protected DDLambda(Context cxt, String xrayTraceInfo) {
        this.tracing = new Tracing(xrayTraceInfo);
        this.enhanced = checkEnhanced();
        recordEnhanced(INVOCATION, cxt);
        addTraceContextToMDC();
        startSpan(new HashMap<>(), cxt);
    }
```

I attempted to create a new branch for this work, but I'm running into some permission issues with doing so. It was a simple enough fix that I thought I'd just post it here.

**Comment by CurtisEppel** [comment]
Thanks @seanstaley, this worked for me too. I've created a PR.

**Comment by CurtisEppel** [comment]
Thanks @seanstaley, this worked for me! I've created a PR.

**Comment by agocs** [comment]
This is fixed on 0.2.2 and greater. Sorry for the hassle!
